kindtest.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+set -o errexit
+
+if [[ -z "$WORKSPACE" ]]; then
+  kinddir="/scratch/$USER/kindtest"
+else
+  kinddir="${WORKSPACE}/logdir/${BUILD_TAG}"
+fi
+mkdir -m777 -p "${kinddir}"
+export RESULT_ROOT="${kinddir}/wl_k8s_test_results"
+if [ -d "$RESULT_ROOT" ]; then
+  rm -Rf "$RESULT_ROOT/*"
+else
+  mkdir -m777 "$RESULT_ROOT"
+fi
+
+export PV_ROOT="${kinddir}/k8s-pvroot"
+if [ -d "$PV_ROOT" ]; then
+  rm -Rf "$PV_ROOT/*"
+else
+  mkdir -m777 "$PV_ROOT"
+fi
+
+echo 'Remove old cluster (if any)...'
+kind delete cluster
+
+# desired cluster name; default is "kind"
+KIND_CLUSTER_NAME="${KIND_CLUSTER_NAME:-kind}"
+
+kind_version=$(kind version)
+kind_network='kind'
+reg_name='kind-registry'
+reg_port='5000'
+case "${kind_version}" in
+  "kind v0.7."* | "kind v0.6."* | "kind v0.5."*)
+    kind_network='bridge'
+    ;;
+esac
+
+echo 'Create registry container unless it already exists'
+running="$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)"
+if [ "${running}" != 'true' ]; then
+  docker run \
+    -d --restart=always -p "${reg_port}:5000" --name "${reg_name}" \
+    registry:2
+fi
+
+reg_host="${reg_name}"
+if [ "${kind_network}" = "bridge" ]; then
+    reg_host="$(docker inspect -f '{{.NetworkSettings.IPAddress}}' "${reg_name}")"
+fi
+echo "Registry Host: ${reg_host}"
+
+echo 'Create a cluster with the local registry enabled in containerd'
+kind_version='v1.15.7@sha256:e2df133f80ef633c53c0200114fce2ed5e1f6947477dbc83261a6a921169488d'
+cat <<EOF | kind create cluster --name "${KIND_CLUSTER_NAME}" --config=-
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+containerdConfigPatches:
+- |-
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${reg_port}"]
+    endpoint = ["http://${reg_host}:${reg_port}"]
+nodes:
+  - role: control-plane
+    image: kindest/node:${kind_version}
+  - role: worker
+    image: kindest/node:${kind_version}
+    extraMounts:
```

Comment:
do we need to expose 6443 or any node ports?

Reply:
no, not on Linux.  The kind documentation says that it's automatic for Linux networking.

Reply:
ok cool